content/articles/2017-12-14-styler-1-0-0.Rmd

```diff
@@ -0,0 +1,234 @@
+
+* "none": Performs no transformation at all.
+* "spaces": Manipulates spacing between token on the same line.
+* "indention": In addition to "spaces", this option also manipulates the indention level.
+* "line_breaks": In addition to "indention", this option also manipulates line breaks.
+* "tokens": In addition to "line_breaks", this option also manipulates tokens.
+
+Hence, `scope < "tokens"` will never change code in the sense that the expression remains the same before and after styling, as only spaces and line breaks between the tokens are modified. styler can internally validate the styling, so you can be sure that it has not broken your code. Although extensively tested, we recommend to inspect the styling result with `scope = "tokens"`, since in this case, an automatic validation is not possible.
+
+Let's now see a few examples. We can limit ourselves to styling just spacing information:
+
+```{r}
+library("styler")
+style_text("a=3; 2", scope = "spaces")
+```
+
+Or, on the other extreme of the scale, styling spaces, indention, line breaks 
+and tokens:
+
+```{r}
+style_text("a=3; 2", scope = "tokens")
+```
+
+In the next example, we want to style spaces and indention, but not line breaks
+and tokens:
+```{r}
+style_text(
+  "a=  function( x) {
+  'x'
+  }", 
+  scope = "indention"
+)
+```
+
+
+### strict
+
+Another option that is helpful to determine the level of 'invasiveness' is `strict`. If set to `TRUE`, spaces and line breaks before or after tokens are set to either zero or one. However, in some situations, this might be undesirable (so we set `strict = FALSE`), as the following example shows:
+
+```{r}
+style_text(
+  "data_frame(
+     small  = 2 ,
+     medium = 4,#comment without space
+     large  =6
+   )", 
+  strict = FALSE
+)
+```
+
+We prefer to keep the equal sign after "small", "medium" and large aligned, so we set `strict = FALSE` to set spacing to *at least* one around `=`.
+
+Also, with `scope >= "line_breaks"`, `strict = TRUE` will break the lines in the following call, whereas `strict = FALSE` will leave them as is:
+```{r}
+style_text(
+  "do_a_long_and_complicated_fun_cal('which', has, way, to, 
+                              'and longer then lorem ipsum in its full length'
+                              )"
+)
+```
+
+
+In addition, the `strict` option does also adds braces (if `scope = "tokens"`) to conditional statements if they are multi-line:
+```{r}
+style_text(
+  "if (set_q(x))
+    3 else 
+    5",
+  strict = TRUE
+)
+```
+
+
+### math token spacing
+
+styler can identify and handle unary operators and other math tokens:
+
+```{styler}
+1++1-1-1/2
+```
+
+This is tidyverse style. However, styler offers very granular control for math token spacing. Assuming you like spacing around `+` and `-`, but not around `/` and `*` and `^`. This can be achieved as follows:
+```{r}
+style_text(
+  "1++1/2*2^2", 
+  math_token_spacing = specify_math_token_spacing(zero = c("'/'", "'*'", "'^'"))
+)
+```
```

Comment:
I am most interested in spacing around `=` and `,` honestly! Would love to see that demonstrated. Meaning:
`f(..., why="do",people="do this?", ...)`
vs
`f(..., why = "do", people = "do this?", ...)`

Reply:
Ok, I added such an example.